src/pages/Onboarding/index.tsx

```diff
@@ -0,0 +1,314 @@
+    };
+
+    try {
+      await api.patch("/users/profile", payload);
+      showDialog(true, "Profile updated");
+      reset(raw);
+      setTimeout(() => navigate("/explore"), 2600);
+    } catch (err) {
+      console.error(err);
+      showDialog(false, "Failed to update profile");
+    }
+  };
+
+  // Prevent interaction with the rest of the page when dialog is open
+  React.useEffect(() => {
+    if (open) {
+      document.body.style.pointerEvents = "none";
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.pointerEvents = "";
+      document.body.style.overflow = "";
+    }
+    return () => {
+      document.body.style.pointerEvents = "";
+      document.body.style.overflow = "";
+    };
+  }, [open]);
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      setImagePreview(URL.createObjectURL(file));
+    }
+  };
+
+  return (
+    <div className="flex min-h-screen items-center justify-center bg-neutral-50 p-6">
+      <Card className="w-full max-w-xl bg-white shadow-xl border border-neutral-200 p-6 rounded-lg">
+        <CardHeader>
+          <h1 className="text-2xl font-bold text-neutral-900 text-center">
+            Complete your profile
+          </h1>
+        </CardHeader>
+        <CardContent>
+          <form onSubmit={handleSubmit(onSubmit)} className="space-y-6">
+            {/* Display Name */}
+            <div className="space-y-1">
+              <label
+                className="text-sm font-medium text-neutral-700"
+                htmlFor="full_name"
+              >
+                Display Name <span className="text-rose-500">*</span>
+              </label>
+              <Input
+                id="full_name"
+                placeholder="your fullname
+                "
+                style={{ color: "#6b7280" }}
+                {...register("full_name", { required: true, maxLength: 80 })}
+                className="w-full px-4 py-2 rounded-lg border border-neutral-300 focus:outline-none focus:ring-2 focus:ring-blue-500"
+              />
+              {errors.full_name && (
+                <p className="text-xs text-rose-500">
+                  Display name is required
+                </p>
+              )}
+            </div>
+
+            {/* Username */}
+            <div className="space-y-1">
+              <label
+                className="text-sm font-medium text-neutral-700"
+                htmlFor="username"
+              >
+                Username <span className="text-rose-500">*</span>
+              </label>
+              <Input
+                id="username"
+                placeholder="your username"
+                style={{ color: "#6b7280" }}
+                {...register("username", { required: true, minLength: 3 })}
+                className="w-full px-4 py-2 rounded-lg border border-neutral-300 focus:outline-none focus:ring-2 focus:ring-blue-500"
```

Comment:
Username must be unique + follows this pattern: 
Character limits:

Minimum: 3 characters
Maximum: 20 characters


Allowed characters:

Lowercase letters (a-z)
Numbers (0-9)
Underscores (_)
Hyphens (-)


Restrictions:

No uppercase letters (for consistency)
No spaces
No special characters (except underscore and hyphen)
No diacritics/accents (không dấu)

put this prompt to your bot

Reply:
if this bug is not resolved, there will be this error: 
![image](https://github.com/user-attachments/assets/beaf75ad-ad4b-4e71-9ac8-6ff2a777f63a)

Reply:
I suggest showing an error message that the username is taken and asks user to choose a different username